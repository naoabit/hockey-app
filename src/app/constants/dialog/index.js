export const ADD_PLAYER_DIALOG_ID = 1;

export default {
    ADD_PLAYER_DIALOG_ID,
};
