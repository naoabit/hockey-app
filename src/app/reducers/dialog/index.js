import { createSlice } from '@reduxjs/toolkit';

const initialState = {
    show: false,
    id: '',
    data: {},
};

const dialogSlice = createSlice({
    name: 'dialog',
    initialState,
    reducers: {
        showDialog: (state) => {
            state.show = true;
        },
        hideDialog: () => initialState,
        setDialog: (state, action) => {
            state.id = action.payload.id;
            state.data = action.payload.data;
        },
    },
});

export const { showDialog, hideDialog, setDialog } = dialogSlice.actions;

export default dialogSlice.reducer;
