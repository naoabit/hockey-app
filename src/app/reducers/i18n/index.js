import { createSlice } from '@reduxjs/toolkit';

const initialState = {
    locale: 'en',
    translations: null,
};

const LOAD_TRANSLATION = '@@i18n/LOAD_TRANSLATIONS';
const SET_LOCALE = '@@i18n/SET_LOCALE';

const i18nSlice = createSlice({
    name: 'i18n',
    initialState,
    extraReducers: {
        [SET_LOCALE]: (state, action) => {
            state.locale = action.locale;
        },
        [LOAD_TRANSLATION]: (state, action) => {
            state.translations = action.translations;
        },
    },
});

export default i18nSlice.reducer;
