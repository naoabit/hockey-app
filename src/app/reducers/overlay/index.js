import { createSlice } from '@reduxjs/toolkit';

const initialState = {
    show: false,
};

const overlaySlice = createSlice({
    name: 'overlay',
    initialState,
    reducers: {
        show: (state) => {
            state.show = true;
        },
        hide: (state) => {
            state.show = false;
        },
    },
});

export const { show, hide } = overlaySlice.actions;

export default overlaySlice.reducer;
