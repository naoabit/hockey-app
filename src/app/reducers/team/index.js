import { createSlice } from '@reduxjs/toolkit';

const initialState = {
    year: 2021,
    coach: '',
    players: [
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Jean', lastName: 'Dujardin', number: 5, position: 'left', isCaptain: false,
        },
        {
            name: 'Scarlett', lastName: 'Johanson', number: 13, position: 'right', isCaptain: false,
        },
        {
            name: 'Kristen', lastName: 'Stewart', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
        {
            name: 'Johnny', lastName: 'Depp', number: 3, position: 'centre', isCaptain: false,
        },
    ],
};

const teamSlice = createSlice({
    name: 'team',
    initialState,
    reducers: {},
    extraReducers: {},
});

export default teamSlice.reducer;
