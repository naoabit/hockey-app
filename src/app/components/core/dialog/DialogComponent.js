import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import PropTypes from 'prop-types';

import { hideDialog } from '@app/reducers/dialog';
import styles from './_scss/dialog.module.scss';

const storePropsSelector = (state) => ({
    show: state.dialog.show,
    overlayShown: state.overlay.show,
});

const DialogComponent = (props) => {
    const { overlayShown, show } = useSelector(storePropsSelector);
    const dispatch = useDispatch();

    useEffect(() => {
        if (!overlayShown) {
            dispatch(hideDialog());
        }
    }, [overlayShown]);

    if (!show) {
        return null;
    }

    return (
        <div className={styles.dialog}>
            {props.children}
        </div>
    );
};

DialogComponent.propTypes = {
    children: PropTypes.node.isRequired,
};

export default DialogComponent;
