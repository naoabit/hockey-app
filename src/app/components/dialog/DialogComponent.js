import React from 'react';
import { useSelector } from 'react-redux';

import { ADD_PLAYER_DIALOG_ID } from '@app/constants/dialog';
import DialogCore from '@app/components/core/dialog/DialogComponent';
import AddPlayerDialog from '@app/components/addPlayerDialog/AddPlayerDialogComponent';

const MAP_DIALOGS = [
    {
        id: ADD_PLAYER_DIALOG_ID,
        children: <AddPlayerDialog/>,
    },
];

const storePropsSelector = ({ dialog }) => ({
    dialogId: dialog.id,
});

const Dialog = () => {
    const { dialogId } = useSelector(storePropsSelector);
    const dialog = MAP_DIALOGS.find((d) => d.id === dialogId);
    if (!dialog) {
        return null;
    }

    return (
        <DialogCore
            children={dialog.children}
        />
    );
};

export default Dialog;
