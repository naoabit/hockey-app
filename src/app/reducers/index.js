import i18n from './i18n';
import dialog from './dialog';
import overlay from './overlay';
import team from './team';

export default {
    i18n,
    dialog,
    overlay,
    team,
};
