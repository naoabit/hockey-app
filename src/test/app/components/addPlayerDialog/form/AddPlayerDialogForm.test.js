import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AddPlayerForm from '@app/components/addPlayerDialog/form/AddPlayerDialogFormComponent';
import { setDialog } from '@app/reducers/dialog';

jest.mock('@app/components/core/formInput/checkbox/FormInputCheckboxComponent', () => 'FormInputCheckbox');
jest.mock('@app/components/core/formInput/text/FormInputTextComponent', () => 'FormInputText');
jest.mock('@app/components/core/formInput/number/FormInputNumberComponent', () => 'FormInputNumber');
jest.mock('@app/reducers/dialog');
jest.mock('react-redux');

describe('<AddPlayerForm />', () => {
    const store = {
        dialog: {
            id: 1,
            data: {
                isCaptain: true,
            },
        },
    };
    useSelector.mockImplementationOnce((fn) => fn(store));
    useDispatch.mockImplementation(() => jest.fn());
    setDialog.mockImplementation(() => jest.fn());

    const wrapper = shallow(
        <AddPlayerForm />,
    );

    it('should render component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.props().className).toEqual('form');
        expect(wrapper.children().length).toEqual(5);

        expect(wrapper.childAt(0).is('FormInputText')).toBe(true);
        expect(wrapper.childAt(1).is('FormInputText')).toBe(true);
        expect(wrapper.childAt(2).is('FormInputNumber')).toBe(true);
        expect(wrapper.childAt(3).is('FormInputText')).toBe(true);
        expect(wrapper.childAt(4).is('FormInputCheckbox')).toBe(true);
    });
    
    it('should dispatch setDialog', () => {
        wrapper.childAt(0).props().onChange({ name: 'Dope name', isCaptain: false });
        expect(setDialog).toHaveBeenCalledWith({
            id: 1,
            data: {
                isCaptain: false,
                name: 'Dope name',
            },
        });
    });
    
});
