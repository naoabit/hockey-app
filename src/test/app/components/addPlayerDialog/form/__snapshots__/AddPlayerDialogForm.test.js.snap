// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPlayerForm /> should render component 1`] = `
<div
  className="form"
>
  <FormInputText
    className="formInput"
    inputKey="name"
    key="name"
    label="addPlayerForm.input.name"
    onChange={[Function]}
  />
  <FormInputText
    className="formInput"
    inputKey="lastName"
    key="lastName"
    label="addPlayerForm.input.lastName"
    onChange={[Function]}
  />
  <FormInputNumber
    className="formInput"
    inputKey="number"
    key="number"
    label="addPlayerForm.input.number"
    onChange={[Function]}
  />
  <FormInputText
    className="formInput"
    inputKey="position"
    key="position"
    label="addPlayerForm.input.position"
    onChange={[Function]}
  />
  <FormInputCheckbox
    className="checkboxInput formInput"
    inputKey="isCaptain"
    key="isCaptain"
    label="addPlayerForm.input.isCaptain"
    onChange={[Function]}
    value={true}
  />
</div>
`;
